scipy/optimize/optimize.py

```diff
@@ -974,14 +993,36 @@ def check_grad(func, grad, x0, *args, **kwargs):
     >>> from scipy.optimize import check_grad
     >>> check_grad(func, grad, [1.5, -1.5])
     2.9802322387695312e-08
+    >>> rng = np.random.default_rng()
+    >>> check_grad(func, grad, [1.5, -1.5], 
+    ...             direction='random', seed=rng)
+    2.9802322387695312e-08
 
     """
-    step = kwargs.pop('epsilon', _epsilon)
-    if kwargs:
-        raise ValueError("Unknown keyword arguments: %r" %
-                         (list(kwargs.keys()),))
-    return sqrt(sum((grad(x0, *args) -
-                     approx_fprime(x0, func, step, *args))**2))
+    step = epsilon
+    x0 = np.asarray(x0)
+    
+    def g(w, func, x0, v, *args):
+        return func(x0 + w*v, *args)
+
+    if direction == 'random':
+        random_state = check_random_state(seed)
+        v = random_state.normal(0, 1, size=(x0.shape))
```

Comment:
It might make sense to normalize the random direction to a unit vector (perhaps we ought to do the same in jax, I now realize). It might not matter much in practice – I haven't checked. In principle, it seems that this would make the meaning of the step parameter (`epsilon`) consistent with the non-random (`'all'`) case.

Reply:
I can try this in a follow up PR. Any thoughts, @rgommers @shoyer ? I have only one concern that `eps*v` will become smaller in absolute value if we normalise `v` (which might increase inaccuracies due to precision issues). May be I am just overthinking and it's not a big issue since we will be reducing the order from 1e-9 to 1e-10.